torch/csrc/expand_utils-inl.h

```diff
@@ -0,0 +1,175 @@
+template <>
+THFloatTensor *newForExpand(LIBRARY_STATE_TYPE_NOARGS) {
+  return THFloatTensor_new();
+}
+
+template <>
+THDoubleTensor *newForExpand(LIBRARY_STATE_TYPE_NOARGS) {
+  return THDoubleTensor_new();
+}
+
+template <>
+THHalfTensor *newForExpand(LIBRARY_STATE_TYPE_NOARGS) {
+  return THHalfTensor_new();
+}
+
+template <>
+THByteTensor *newForExpand(LIBRARY_STATE_TYPE_NOARGS) {
+  return THByteTensor_new();
+}
+
+template <>
+THCharTensor *newForExpand(LIBRARY_STATE_TYPE_NOARGS) {
+  return THCharTensor_new();
+}
+
+template <>
+THShortTensor *newForExpand(LIBRARY_STATE_TYPE_NOARGS) {
+  return THShortTensor_new();
+}
+
+template <>
+THIntTensor *newForExpand(LIBRARY_STATE_TYPE_NOARGS) {
+  return THIntTensor_new();
+}
+
+template <>
+THLongTensor *newForExpand(LIBRARY_STATE_TYPE_NOARGS) {
```

Comment:
We expand on the CPU because the CPU copy methods have THCState if the source is on the GPU.